src/components/CommandBar/CommandsList/index.tsx

```diff
@@ -0,0 +1,156 @@
+  setIsOpen,
+} from 'src/redux/slices/commandBar';
+import { resolveUrlBySearchNavigationType } from 'src/utils/navigation';
+import { SearchNavigationResult } from 'types/SearchNavigationResult';
+
+interface Props {
+  isPreInput?: boolean;
+  navigationItems: SearchNavigationResult[];
+}
+
+const CommandsList: React.FC<Props> = ({ navigationItems, isPreInput = false }) => {
+  const numberOfItems = navigationItems.length;
+  const [scrollToSelectedItem, selectedItemRef]: [() => void, RefObject<HTMLLIElement>] =
+    useScroll(SMOOTH_SCROLL_TO_CENTER);
+  const [selectedItemIndex, setSelectedItemIndex] = useState(() => (!numberOfItems ? null : 0));
+  const router = useRouter();
+  const dispatch = useDispatch();
+  const onUpKeyClicked = useCallback(
+    (event: KeyboardEvent) => {
+      // prevent the input's cursor to jump to the end of the input value.
+      event.preventDefault();
+      setSelectedItemIndex((prevSelectedItemIndex) => prevSelectedItemIndex - 1);
+      scrollToSelectedItem();
+    },
+    [scrollToSelectedItem],
+  );
+  const onDownKeyClicked = useCallback(
+    (event: KeyboardEvent) => {
+      // prevent the input's cursor to jump to the beginning of the input value.
+      event.preventDefault();
+      setSelectedItemIndex((prevSelectedItemIndex) => prevSelectedItemIndex + 1);
+      scrollToSelectedItem();
+    },
+    [scrollToSelectedItem],
+  );
+  const navigateToItem = useCallback(
+    (navigationItem: SearchNavigationResult) => {
+      router
+        .push(resolveUrlBySearchNavigationType(navigationItem.resultType, navigationItem.key))
+        .then(() => {
+          dispatch({ type: addRecentNavigation.type, payload: navigationItem });
+          dispatch({ type: setIsOpen.type, payload: false });
+        });
+    },
+    [dispatch, router],
+  );
+  useHotkeys(
+    'up',
+    onUpKeyClicked,
+    {
+      enabled: numberOfItems && selectedItemIndex !== 0,
+      enableOnTags: ['INPUT'],
+    },
+    [scrollToSelectedItem],
+  );
+  useHotkeys(
+    'down',
+    onDownKeyClicked,
+    {
+      enabled: numberOfItems && selectedItemIndex !== numberOfItems - 1,
+      enableOnTags: ['INPUT'],
+    },
+    [scrollToSelectedItem],
+  );
+  useHotkeys(
+    'enter',
+    () => {
+      navigateToItem(navigationItems[selectedItemIndex]);
+    },
+    { enabled: selectedItemIndex !== null, enableOnTags: ['INPUT'] },
+    [navigateToItem, navigationItems, selectedItemIndex],
+  );
+  const onRemoveItemClicked = (event: MouseEvent<Element>, navigationItemKey: number | string) => {
+    // to not allow the event to bubble up to the parent container
+    event.stopPropagation();
+    dispatch({ type: removeRecentNavigation.type, payload: navigationItemKey });
+  };
+
+  if (!numberOfItems) {
+    return (
+      <p className={styles.noResult}>
+        {isPreInput ? 'No recent navigations' : 'No navigation results'}{' '}
+      </p>
+    );
+  }
+
+  const onItemHover = (itemIndex: number) => {
```

Comment:
nit: should we call this highlight/select instead of hover? hover usually indicates mouse hovering